tree/dataframe/src/RLoopManager.cxx

```diff
@@ -475,8 +536,22 @@ void RLoopManager::RunTreeProcessorMT()
 void RLoopManager::RunTreeReader()
 {
    TTreeReader r(fTree.get(), fTree->GetEntryList());
-   if (0 == fTree->GetEntriesFast())
+
+   // If `GetEntries` is not called, in case of chains, `SetEntriesRange` might not properly indicate that a
+   // start entry is too large, e.g. when start=last entry
+   // The `GetEntries(true)` call can be substituted with the more efficient GetEntries(false) or GetEntriesFast()
+   // once `https://github.com/root-project/root/issues/10774` is resolved.
+   const auto nEntries = r.GetEntries(/*force=*/true);
+
+   if (fStartEntry == fEndEntry || 0 == nEntries)
```

Comment:
I'm very wary of increasing the number of places from which we trigger the opening of all input files. 

if `GetEntriesFast()` was enough before, `GetEntries(false)` should also suffice.

if `GetEntriesFast()` was _not_ enough before, and you are improving RDF's diagnostics here, this seems unrelated to the introduction of `RDatasetSpec` and we should tackle the problem in a different PR.

Reply:
I see. A fix is: if the `REntryRange` was specified =>  open all files here (which `SetEntryRange` would do). Otherwise, just open 1 file.

In this way the old behaviour of the RLoopManager would remain unchanged.

Reply:
> which `SetEntryRange` would do

where does it open all files?

Reply:
I just answered here: https://github.com/root-project/root/issues/10774#issuecomment-1171122132

So, I think there are 2 solutions as of now:
* If no range is specified, call `GetEntries(false)` and subsequently do not call SetEntriesRange --> very easy to implement, does not change existent RLoopManager handling.
* Call SetEntriesRange in later stage, or/and maybe need special function to apply the range directly in the event loop --> I need more time to make that work, again that does not affect current RLoopManager's handling.